src/panel/direction/DirectionPanel.jsx

```diff
@@ -334,13 +334,14 @@ export default class DirectionPanel extends React.Component {
       {isMobile => isMobile
         ? <Fragment>
           {!activePreviewRoute && <div className="direction-panel" ref={this.setMarginTop}>
-            <Flex
+            {(routes.length === 0 && !isLoading) && <Flex
```

Comment:
It would be more robust to use the same display condition as the result panel, i.e. having filled up `origin` and `destination`. That way the change will happen in the same render pass.
It would also hide the title in case on an error, which is also more consistent: when you go from a 'car' result to 'public transport' and the latter doesn't have results, you don't need to see the title again.

Reply:
you r right, thanks :+1: